consensus/polybft/stake_manager.go

```diff
@@ -109,45 +106,32 @@ func (s *stakeManager) PostBlock(req *PostBlockRequest) error {
 	s.logger.Debug("Gotten transfer (stake changed) events from logs on block",
 		"eventsNum", len(events), "block", req.FullBlock.Block.Number())
 
+	for _, e := range events {
+		s.logger.Debug("Transfer event", "From", e.From, "To", e.To, "Value", e.Value)
+	}
+
 	fullValidatorSet, err := s.state.StakeStore.getFullValidatorSet()
 	if err != nil {
 		return err
 	}
 
 	stakeMap := fullValidatorSet.Validators
 
-	updatedValidatorsStake := make(map[types.Address]struct{}, 0)
+	s.logger.Debug("full validator set before", "block", fullValidatorSet.BlockNumber, "data", stakeMap)
 
 	for _, event := range events {
 		if event.IsStake() {
-			updatedValidatorsStake[event.To] = struct{}{}
+			// then this amount was minted To validator address
+			stakeMap.addStake(event.To, event.Value)
 		} else if event.IsUnstake() {
-			updatedValidatorsStake[event.From] = struct{}{}
+			// then this amount was burned From validator address
+			stakeMap.removeStake(event.From, event.Value)
 		} else {
+			// this should not happen, but lets log it if it does
 			s.logger.Debug("Found a transfer event that represents neither stake nor unstake")
 		}
 	}
 
-	// this is a temporary solution (a workaround) for a bug where amount
-	// in transfer event is not correctly generated (unknown 4 bytes are added to begging of Data array)
-	if len(updatedValidatorsStake) > 0 {
-		provider, err := s.blockchain.GetStateProviderForBlock(req.FullBlock.Block.Header)
-		if err != nil {
-			return err
-		}
-
-		systemState := s.blockchain.GetSystemState(provider)
-
-		for a := range updatedValidatorsStake {
-			stake, err := systemState.GetStakeOnValidatorSet(a)
```

Comment:
Remove the `GetStakeOnValidatorSet` function from `systemState` interface, `SystemStateImpl`, and `systemStateMock`.